### PR TITLE
Wf new plugin iface

### PR DESCRIPTION
**Pull request by surindersinghp**

### CODE REVIEW

**Comment by kumare3** [comment]
why are there so many commits?

**Comment by surindersinghp** [comment]
why are there so many commits?
>> not sure, something to do with merge. Let me recreate to make it cleaner.

File: pkg/apis/flyteworkflow/v1alpha1/node_status.go [hunk 1]
```diff
@@ -80,20 +80,16 @@ func (s *DynamicNodeStatus) Equals(o *DynamicNodeStatus) bool {
...
-	s.Phase = phase
-}
+const (
+	WorkflowNodePhaseNone WorkflowNodePhase = iota
+	WorkflowNodePhaseExecuting
+)
 
 type WorkflowNodeStatus struct {
-	WorkflowName string `json:"name"`
+	WorkflowName string            `json:"name"`
```
Comment (kumare3) [inline pkg/apis/flyteworkflow/v1alpha1/node_status.go]: do we need to store the workflowname ? or is it just for debugging. If so we can generate it using kubectl flyte

File: pkg/apis/flyteworkflow/v1alpha1/node_status.go [hunk 2]
```diff
@@ -80,20 +80,16 @@ func (s *DynamicNodeStatus) Equals(o *DynamicNodeStatus) bool {
...
-func (s SubWorkflowNodeStatus) GetPhase() WorkflowPhase {
-	return s.Phase
-}
+type WorkflowNodePhase int
 
-func (s *SubWorkflowNodeStatus) SetPhase(phase WorkflowPhase) {
-	s.Phase = phase
-}
+const (
+	WorkflowNodePhaseNone WorkflowNodePhase = iota
```
Comment (kumare3) [inline pkg/apis/flyteworkflow/v1alpha1/node_status.go]: undefined?

File: pkg/apis/flyteworkflow/v1alpha1/node_status.go (hunk 1 above)
Comment (surindersinghp) [reply pkg/apis/flyteworkflow/v1alpha1/node_status.go]: removed.

File: pkg/apis/flyteworkflow/v1alpha1/node_status.go (hunk 2 above)
Comment (surindersinghp) [reply pkg/apis/flyteworkflow/v1alpha1/node_status.go]: fixed

**Comment by surindersinghp** [comment]
merging to refactor

**Pull request merged**